hypex/dataset/dataset.py

```diff
@@ -319,6 +320,40 @@ def _check_other_dataset(self, other):
             raise ConcatDataError(type(other))
         if type(other._backend) is not type(self._backend):
             raise ConcatBackendError(type(other._backend), type(self._backend))
+        
+    def astype(self, dtype: Dict[str, DefaultRoleTypes], errors: Literal["raise", "ignore"] = "raise") -> "Dataset":
+        """
+        Change the data type of one or more columns.
+        
+        Parameters:
+        - dtype: Dictionary where keys are column names and values are target types.
+        - errors: If 'raise', raises an error on invalid types; if 'ignore', skips invalid types.
+        
+        Returns:
+        - A new Dataset object with the specified data types applied.
+        """
+        new_data = self._backend.data.copy()
+        new_roles = deepcopy(self.roles)
+        
+        for col, target_type in dtype.items():
+            if col not in new_data.columns:
+                if errors == "raise":
+                    raise KeyError(f"Column '{col}' does not exist in the Dataset.")
+                elif errors == "ignore":
+                    warnings.warn(f"Column '{col}' does not exist. Skipping conversion.", category=UserWarning)
+                    continue
+            
+            try:
```

Comment:
rework without use of try - except, level of this code is same as if (above) and else (below)